### PR TITLE
KAFKA-2327; broker doesn't start if config defines advertised.host but not advertised.port

### PR DESCRIPTION
**Pull request by granders**
Added unit tests as well. These fail without the fix, but pass with the fix.

### CODE REVIEW

**Comment by gwenshap** [comment]
LGTM. Thanks for the catch. I'm running tests locally to validate and if all passes I'll (try to) merge.

File: core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala [hunk 1]
```diff
@@ -170,8 +170,41 @@ class KafkaConfigTest extends JUnit3Suite {
...
   }
+    
+  @Test
+  def testAdvertisePortDefault() {
+    val advertisedHostName = "routable-host"
+    val port = "9999"
+
+    val props = TestUtils.createBrokerConfig(0, TestUtils.MockZkConnect)
+    props.put(KafkaConfig.AdvertisedHostNameProp, advertisedHostName)
+    props.put(KafkaConfig.PortProp, port.toString)
```
Comment (eribeiro) [inline core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala]: as `port` is String you don't need to call `toString` on it.

File: core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala [hunk 2]
```diff
@@ -170,8 +170,41 @@ class KafkaConfigTest extends JUnit3Suite {
...
+  }
+  
+  @Test
+  def testAdvertiseHostNameDefault() {
+    val hostName = "routable-host"
+    val advertisedPort = "9999"
+
+    val props = TestUtils.createBrokerConfig(0, TestUtils.MockZkConnect)
+    props.put(KafkaConfig.HostNameProp, hostName)
+    props.put(KafkaConfig.AdvertisedPortProp, advertisedPort.toString)
```
Comment (eribeiro) [inline core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala]: Same as above, as `advertisedPort` is String you don't need to call `toString` on it.

**Comment by eribeiro** [comment]
In fact, those new test methods came from `testAdvertiseConfigured`, right? So it would make sense to change `props.put(KafkaConfig.AdvertisedPortProp, advertisedPort.toString)` there too. ;)

File: core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala (hunk 2 above)
Comment (gwenshap) [inline core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala]: advertisedPort is an int:
  val advertisedPort: java.lang.Integer = Option(getInt(KafkaConfig.AdvertisedPortProp)).getOrElse(port)

**Comment by granders** [comment]
@eribeiro Good point :)
@gwenshap Thanks for the quick review - have you seen this? @ijuma posted merge instructions for github here: https://cwiki.apache.org/confluence/display/KAFKA/Patch+submission+and+review#Patchsubmissionandreview-MergingGitHubPullRequests

And also this:
https://cwiki.apache.org/confluence/display/KAFKA/Contributing+Code+Changes

**Comment by ijuma** [comment]
Note: the merge script is attached to the JIRA issue, but it's still waiting for someone to merge it.

**Comment by ijuma** [comment]
Issue link https://issues.apache.org/jira/browse/KAFKA-2187

**Comment by gwenshap** [comment]
@granders, can you change the PR name to something that matches our standard? It looks like the merge script automatically uses the PR name as the commit comment, so we need something like:
"KAFKA-2327; broker doesn't start if config defines advertised.host but not advertised.port"

Or @ijuma can modify the script to allow editing the commit message :)

**Comment by granders** [comment]
@gwenshap Sounds good - I changed the PR title - does that do the trick?

**Comment by gwenshap** [comment]
Yes! Should be merged now.

**Comment by granders** [comment]
Awesome, thanks!

**Pull request closed without merging**